### ISSUE TITLE
[question] Overriding default keystrokes for grid interaction. 

### ISSUE THREAD

**Issue by tielushko** [mui/mui-x#5551]
### Order ID 💳

31986

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

Hello MUI Team,

I was looking for a way to override the default keyboard combination needed to achieve a certain behavior. 

For example: 
For selection, I would like to click the row to select it, and I could click it to unselect it without the need to press CTRL. 

I would greatly appreciate your support!

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by DanailH** [comment]
Hi, @tielushko thanks for raising this!
I don't think we support that type of extension. I'm interested in knowing what your use case is for overwriting the keyboard combinations? When we were deciding on them we took into account what users are already used to from different data tables.

**Comment by flaviendelangle** [comment]
This makes me think off https://github.com/mui/mui-x/issues/1302

**Comment by flaviendelangle** [comment]
We closed it recently because we felt like it was a very heavy mechanism for something we did not felt any traction for

I still don't think the use cases justify something that generic.

**Comment by tielushko** [comment]
@DanailH The reason mostly being picky clients 😆 They don't like to have to press the extra CTRL key when trying to unselect the row, and would love to have it just be a click.

I think for now, I resorted to pas disableSelection to the Data Grid, and overriding the onRowClick event with a custom function that will use the apiRef to programmatically select, unselect rows. Still wasn't able to mimick the sort of selection with shift and the rows in between, but it is not a popular use case, and can be ommitted in favor of having a feature described

**Comment by tielushko** [comment]
@flaviendelangle Thank you for finding the related issue. I understand the heavy lift it would be for such simple feature. Please feel free to close the issue as needed, as I have found the workaround to the issue.

Thanks @DanailH and @flaviendelangle !

**Comment by flaviendelangle** [comment]
I think your use-case is valid.
My question is more about the solution we should give.

Could you give me your workaround ?
Because you should be able to do it though the event system, and maybe we can make it easier by improving this event system rather than introducing the new concept described in #1302

**Comment by tielushko** [comment]
@flaviendelangle Here's a codesanbox. Hope you'll find that helpful.

https://codesandbox.io/s/unselect-row-onclick-w-o-ctrl-8h70nw

**Comment by flaviendelangle** [comment]
OK your doing pretty much what I had in mind

For this behavior we have a `disableSelectionOnClick` that makes it easy to disable the base behavior.
Otherwise, most default behaviors can be disabled using `muiDefaultPrevented` ([doc](https://mui.com/x/react-data-grid/events/#disabling-the-default-behavior)).
But it will only works if the disabling is done before the default beharior runs. Which is not something that is always easy to do, so our doc could probably be improved on that point.

**Comment by tielushko** [comment]
Yeah the reason I disabled it with a prop rather than default prevented was because when I went with the docs it still seemed to trigger on keystrokes. 

But this is general idea I went with for my request

**Comment by shetamitesh** [comment]
I also need the same functionality. In my case, I am using Datagrid for Stock Watchlist and I want to enable users to delete stocks with Delete key. Also is there a way I can edit default functionality of given keystrokes?
